dcap/types/src/collateral.rs

```diff
@@ -75,13 +76,18 @@ impl From<FromUtf8Error> for Error {
 ///
 /// The certificate chains and CRLs are documented in
 /// <https://api.trustedservices.intel.com/documents/Intel_SGX_PCK_Certificate_CRL_Spec-1.5.pdf>
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub struct Collateral {
+    #[serde(with = "certificate_list")]
```

Comment:
I'm not fond of how manual this feels for serde, but the `X509_cert` types didn't derive serde, we could wrap them in NewTypes but we'd still be stuck manually implementing the de/serialize.  IFF we wanted to keep these as vecs of DER bytes on the Collateral we could get it for free, but then we'd need to convert to/from the Certificate and CertificateList types which is fallible.